File: 2_semester/Laba_6/task1_3.cpp
#include <iostream>
#include <vector>
using namespace std;

bool is_main_simetr(vector<vector<int>>& vec) {
	int n = vec.size();
	for (int i = 0; i < n - 1; i++)
		for (int j = i + 1; j < n; j++)
			if (vec[i][j] != vec[j][i])
				return	false;
	return	true;
}

bool is_nomain_simetr(vector<vector<int>>& vec) {
	int n = vec.size();
	for (int i = 0; i < n - 1; i++)
		for (int j = 0; j < n - (i + 1); j++)
			if (vec[i][j] != vec[n - 1 - j][n - 1 - i])
				return	false;
	return	true;
}

int main()
{
	int n = 3;
	cin >> n;
	vector<vector<int>> main_vector(n);
	for (auto& line : main_vector)
		for (auto i : main_vector)
			line.push_back(rand() % 5 + 1);

	for (auto line : main_vector) {
		for (auto item : line)
			cout << item << ' ';
		cout << endl;
	}

	is_main_simetr(main_vector)  ? cout << "YES" << endl : cout << "NO" << endl;
	is_nomain_simetr(main_vector)? cout << "YES" << endl : cout << "NO" << endl;

	return 0;
}

File: 2_semester/Laba_6/task1_3.py
import random

def is_main_simetr(vec):
	n = len(vec)
	for i in range(n-1):
		for j in range(i+1,n):
			if (vec[i][j] != vec[j][i]):
				return	False
	return True

def is_nomain_simetr(vec):
	n = len(vec)
	for i in range(n-1):
		for j in range(n-(i+1)):
			if (vec[i][j] != vec[n - 1 - j][n - 1 - i]):
				return	False
	return	True


n = 3
n = int(input())
main_vector = []
for _ in range(n):
	main_vector.append([int(random.random() * 100) % 5 + 1 for _ in range(n)])

for line in main_vector:
	for item in line:
		print(item, end=' ')
	print()


if is_main_simetr(main_vector):
	print("YES")
else:
	print("NO")

if is_nomain_simetr(main_vector): 
	print("YES")
else:
	print("NO")